### PR TITLE
pkgconf: update to support Conan 2.0 

### PR DESCRIPTION
**Pull request by jcar87**
Specify library name and version:  **pkgconf/all**

### Summary of changes
* Use newer integrations to build (`Meson` and `Mesontoolchain`)
* General update to use tools compatible with Conan 2.0
* Update meson build time requirement to Meson 1.0.0
* Remove `_sharedstatedir` internal property - tested against all supported versions of the recipe and seems to make no difference in the packaged artifacts
* vcvars istantiation is no longer needed as it is now implicitly done by `Meson`/`MesonToolchain`

The test package is slightly refactored for v2:
* Test integration of finding `libexample1.pc` directly during the same step that executes autoreconf+configure on a small autotools project
* Use cmake target generated by CMake deps, rather than legacy integration

### CODE REVIEW

File: recipes/pkgconf/all/conanfile.py [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+
+required_conan_version = ">=1.57.0"
 
 
 class PkgConfConan(ConanFile):
     name = "pkgconf"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://git.sr.ht/~kaniini/pkgconf"
-    topics = ("pkgconf")
     license = "ISC"
```
Comment (StellaSmith) [inline recipes/pkgconf/all/conanfile.py:21]:
```suggestion
    homepage = "https://git.sr.ht/~kaniini/pkgconf"
    topics = ("build", "configuration")
    license = "ISC"
```

Comment (jcar87) [reply recipes/pkgconf/all/conanfile.py:22]: Thanks!! Done :)

**Review by StellaSmith — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]
LGTM! Great job 👏

File: recipes/pkgconf/all/conanfile.py [hunk 2]
```diff
@@ -26,111 +33,94 @@ class PkgConfConan(ConanFile):
...
+            self.options.rm_safe("fPIC")
+       
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
     def validate(self):
-        if hasattr(self, "settings_build") and tools.cross_building(self):
-            raise ConanInvalidConfiguration("Cross-building is not implemented in the recipe")
+        if cross_building(self):
+            raise ConanInvalidConfiguration("Cross-building is not implemented in the recipe, contributions welcome.")
```
Comment (SpaceIm) [inline recipes/pkgconf/all/conanfile.py:58]: But cross-build should be ok with the migration to MesonToolchain.

**Review by SSE4 — APPROVED** [review]

**Pull request merged**